modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportGENCC.pm

```diff
@@ -0,0 +1,325 @@
+    );
+    $self->throw("Missing attrib_type '$code' in variation DB.") unless $id;
+    $atid{$code} = $id;
+  }
+
+  # SQL
+  my $dbq_hgnc_to_ensg = $dbh_core->prepare(q{
+    SELECT g.stable_id
+      FROM gene g
+      JOIN object_xref ox ON ox.ensembl_id=g.gene_id AND ox.ensembl_object_type='Gene'
+      JOIN xref x         ON x.xref_id=ox.xref_id
+      JOIN external_db ed ON ed.external_db_id=x.external_db_id
+    WHERE ed.db_name='HGNC'
+      AND (x.dbprimary_acc=? OR x.dbprimary_acc=CONCAT('HGNC:',?))
+    LIMIT 1
+  });
+  my $dbq_phen_sel = $dbh_var->prepare('SELECT phenotype_id FROM phenotype WHERE description=? LIMIT 1');
+  my $dbq_phen_ins = $dbh_var->prepare('INSERT INTO phenotype (description) VALUES (?)');
+  my $dbq_pf_exists = $dbh_var->prepare(q{
+    SELECT 1 FROM phenotype_feature
+     WHERE phenotype_id=? AND source_id=? AND type='Gene' AND object_id=? LIMIT 1
+  });
+  my $dbq_pf_ins = $dbh_var->prepare(q{
+    INSERT INTO phenotype_feature (phenotype_id, source_id, type, object_id)
+    VALUES (?,?, 'Gene', ?)
+  });
+  my $dbq_pfa_ins = $dbh_var->prepare(q{
+    INSERT INTO phenotype_feature_attrib (phenotype_feature_id, attrib_type_id, value)
+    VALUES (?,?,?)
+  });
+
+  # Read CSV, clean fields, and load rows
+  my $csv = Text::CSV_XS->new({ binary => 1, auto_diag => 1 });
+  open(my $IN, "<:encoding(UTF-8)", $file) or $self->throw("Cannot open $file: $!");
+
+  my $header = $csv->getline($IN) or $self->throw("Empty CSV: $file");
+  $csv->column_names(@$header);
+
+  my ($n_rows, $n_skip, $n_no_gene, $n_dupe, $n_loaded) = (0,0,0,0,0);
+
+  ROW: while (my $r = $csv->getline_hr($IN)) {
+    $n_rows++;
+
+    my $submitter = _clean_text($r->{'submitter_title'}      // '');
+    my $disease   = _clean_text($r->{'disease_title'}        // '');
+    my $uuid      = _clean_text($r->{'uuid'}                 // '');
+    my $class     = _clean_text($r->{'classification_title'} // '');
+    my $moi       = _clean_text($r->{'moi_title'}            // '');
+
+    # Skip excluded submitters (not “unmatched”; do not log)
+    next ROW if $submitter && $skip{$submitter};
+
+    # Skip empty disease (log as unmatched: no_identifier)
+    unless ($disease) {
+      $self->print_logFH(join("\t", "UNMATCHED", "no_identifier", ($uuid//''), ($submitter//''), ($r->{'gene_curie'}//''), ($r->{'gene_symbol'}//''), ($disease//'')) . "\n");
+      next ROW;
+    }
+
+    # Find the gene (HGNC id first, fallback to symbol)
+    my $ensg;
+    my $curie_raw = $r->{'gene_curie'}  // '';
+    my $sym_raw   = $r->{'gene_symbol'} // '';
+
+    if ((my $curie = $r->{'gene_curie'} // '') =~ /^HGNC:(\d+)$/) {
+      # Try both plain number and prefixed HGNC:<number>
+      $dbq_hgnc_to_ensg->execute($1, $1);
+      ($ensg) = $dbq_hgnc_to_ensg->fetchrow_array;
+    }
+    unless ($ensg) {
+      if ($sym_raw) {
+        my $hits = $gene_adaptor ? ($gene_adaptor->fetch_all_by_external_name($sym_raw) || []) : [];
```

Comment:
Around 2000 genes are being saved with the LRG ID. It would be better to use the Ensembl ID for all of them.
You can check how it is done in the G2P import: https://github.com/Ensembl/ensembl-variation/blob/8550e21e23780257888950c754662bc41c0eeba0/modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportG2P.pm#L235-L244 and [Orphanet](https://github.com/Ensembl/ensembl-variation/blob/8550e21e23780257888950c754662bc41c0eeba0/modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/ImportOrphanet.pm#L202-L214).